spec/observables/dom/webSocket-spec.ts

```diff
@@ -416,6 +416,52 @@ describe('webSocket', () => {
       subject.unsubscribe();
     });
 
+    it('should take a serializer', () => {
+      const subject = webSocket<{ data: string}>({
+        url: 'ws://mysocket',
+        serializer: (e: any) => {
+          return e.data + '!';
+        }
+      });
+
+      subject.subscribe();
+
+      const socket = MockWebSocket.lastSocket;
+      socket.open();
+
+
+      ['ahoy', 'yarr', 'shove off'].forEach(x => {
+        subject.next({ data: x });
+        expect(socket.lastMessageSent).to.equal(x + '!');
+      });
+
+      subject.unsubscribe();
+    });
+
+    it('if the serializer fails it should close the connection and rethrow the error', () => {
+      const subject = webSocket<string>({
+        url: 'ws://mysocket',
+        serializer: (e: any) => {
+          throw new Error('I am a bad error');
+        }
+      });
+
+      subject.subscribe({ next: (x: any) => {
+        expect(x).to.equal('this should not happen');
+      }, error: (err: any) => {
+        expect(err).to.equal('this should not happen');
+      } });
+
+      const socket = MockWebSocket.lastSocket;
+      socket.open();
+
+      expect(() => subject.next('weee!')).to.throw('I am a bad error');
```

Comment:
Issue: We wouldn't want this to synchronously throw. No call to `next()` should sync throw, really.

Reply:
I see your point, but at the same time the serializer kicks in on `subject.next()`.

In a way, the serializer option is just a shorthand so that the consumer doesn't need to do this every time:

```ts
subject.next(serialize(message))
```

which obviously would throw synchronously.

Just food for thought whenever you discuss about this.